src/nvim/eval.c

```diff
@@ -14456,11 +14456,11 @@ static void f_setreg(typval_T *argvars, typval_T *rettv)
   char_u      *strregname;
   char_u      *stropt;
   bool append = false;
-  char_u yank_type;
+  MotionType yank_type;
   long block_len;
 
   block_len = -1;
-  yank_type = MAUTO;
+  yank_type = kUnknownMotion;
```

Comment:
This should be something like kMotionTypeUnknown, kMotionTypeChar, kMotionTypeBlock, kMotionTypeLine. I.e. with prefix common for all enum members.